src/lkl/setup.c

```diff
@@ -1285,16 +1285,20 @@ static void* lkl_termination_thread(void* args)
     SGXLKL_VERBOSE("calling lkl_virtio_netdev_remove()\n");
     lkl_virtio_netdev_remove();
 
-    SGXLKL_VERBOSE("calling lkl_sys_halt()\n");
-    res = lkl_sys_halt();
-    if (res < 0)
-    {
-        sgxlkl_fail("LKL halt, %s\n", lkl_strerror(res));
-    }
+    /**
+     * If kernel threads are stuck, this may block indefinitely under
+     * cooperative scheduling.
+     */
+    // SGXLKL_VERBOSE("calling lkl_sys_halt()\n");
```

Comment:
was this comment code left behind by mistake?

Reply:
I would like to have a discussion here if it is the right thing to remove this (at least for now). It may even make sense to leave in the comment, so we don't forget to think of a way to put it back in.

Reply:
I'd rather see an issue or issues opened with a link back to the removal commit.

Reply:
Sure, I can do this once we've decided if this shutdown design makes more sense than the current one.